### PR TITLE
adds a checkbox to exclude archived assets from custom reports

### PR DESCRIPTION
**Pull request by Godmartinz**
# Description
Adds a checkbox to exclude archived assets from custom reports

Fixes # SC-19107

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* PHP version:
* MySQL version
* Webserver version
* OS version

# Checklist:

- [ ] I have read the Contributing documentation available here: https://snipe-it.readme.io/docs/contributing-overview
- [ ] I have formatted this PR according to the project guidelines: https://snipe-it.readme.io/docs/contributing-overview#pull-request-guidelines
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/Http/Controllers/ReportsController.php [hunk 1]
```diff
@@ -642,6 +642,9 @@ public function postCustom(Request $request)
             if (($request->filled('next_audit_start')) && ($request->filled('next_audit_end'))) {
                 $assets->whereBetween('assets.next_audit_date', [$request->input('next_audit_start'), $request->input('next_audit_end')]);
             }
+            if($request->filled('exclude_archived')){
+                $assets->where('archived', '=', 0);
```
Comment (snipe) [inline app/Http/Controllers/ReportsController.php]: I think this wouldn't account for the assets that are not individually marked as archived, but instead belong to an archived status label. 

Maybe check out this query scope:
https://github.com/snipe/snipe-it/blob/3182f251e1b4a529069b8e6473f3f908cc267494/app/Models/Asset.php#L1062-L1067

Comment (Godmartinz) [reply app/Http/Controllers/ReportsController.php]: Accounts for status_id now as well 👍

**Comment by Godmartinz** [comment]
`->notArchived()` is being used now

**Comment by snipe** [comment]
Great! Can you pull the commented line out? Once that's done, it looks good to me.

**Pull request merged**